packages/@react-spectrum/combobox/src/MobileComboBox.tsx

```diff
@@ -402,6 +423,16 @@ function ComboBoxTray(props: ComboBoxTrayProps) {
     lastInputValue.current = inputValue;
   }, [loadingState, inputValue, showLoading]);
 
+  // Need to handle onKeyDown here, Tray's onClose doesn't trigger when hitting escape since useComboBox's escape handler closes the tray
+  let onKeyDown = (e) => {
+    // If there isn't a selected key and allowsCustomValue is true, then we avoid reseting input value so the new custom value persists
+    if (e.key === 'Escape' && !(!state.selectedKey && allowsCustomValue)) {
+      state.resetInputValue();
```

Comment:
I think this could go away also if you call `state.setFocused(false)` when the tray unmounts?

Reply:
We still need this to reset the input value when the user hits escape (unless you are suggesting that we make `shouldCloseOnBlur=true` in our `useComboBox` call?)